### PR TITLE
Fixes issue

### PR DESCRIPTION
**Pull request by hidanielesgit**

### CODE REVIEW

File: include/alp/base/blas0.hpp [hunk 1]
```diff
@@ -403,8 +403,8 @@ namespace alp {
 		template< alp::Descriptor descr, typename OutputType, typename D >
 		class ValueOrIndex< 
 			descr, OutputType, D,
-			typename std::enable_if< std::is_arithmetic< OutputType >::value
-			&& ! std::is_same< D, void >::value >::type 
+			std::enable_if_t< std::is_arithmetic< OutputType >::value
```
Comment (djelovina) [inline include/alp/base/blas0.hpp:406]: wring line break and missing `* = nullptr` at the end

File: include/alp/base/blas0.hpp [hunk 2]
```diff
@@ -429,8 +429,8 @@ namespace alp {
 		template< alp::Descriptor descr, typename OutputType, typename D >
 		class ValueOrIndex<
 			descr, OutputType, D,
-			typename std::enable_if< ! std::is_arithmetic< OutputType >::value
-			&& ! std::is_same< OutputType, void >::value >::type
+			std::enable_if_t< ! std::is_arithmetic< OutputType >::value
```
Comment (djelovina) [inline include/alp/base/blas0.hpp:432]: same as above

File: include/alp/amf-based/matrix.hpp [hunk 3]
```diff
@@ -88,19 +88,19 @@ namespace alp {
 		/** Forward declarations for access functions */
 		template<
 			typename MatrixType,
-			std::enable_if< is_matrix< MatrixType >::value > * = nullptr
+			std::enable_if_t< is_matrix< MatrixType >::value > * = nullptr
```
Comment (djelovina) [inline include/alp/amf-based/matrix.hpp:91]: I think we should use `const` for `nullptr`, is that correct?

File: include/alp/amf-based/storagebasedmatrix.hpp [hunk 4]
```diff
@@ -167,7 +167,7 @@ namespace alp {
 			/** Get the reference to the AMF of a storage-based matrix */
 			template<
 				typename MatrixType,
-				std::enable_if< internal::is_storage_based< MatrixType >::value > *
+				std::enable_if_t< internal::is_storage_based< MatrixType >::value > *
```
Comment (djelovina) [inline include/alp/amf-based/storagebasedmatrix.hpp:170]: missing `= nullptr`

File: include/alp/reference/blas0.hpp [hunk 5]
```diff
@@ -68,12 +68,12 @@ namespace alp {
...
 			const InputType2 &y,
 			const OP &op = OP(),
-			const typename std::enable_if<
+			const std::enable_if_t<
 				is_operator< OP >::value &&
 				!is_object< InputType1 >::value &&
 				!is_object< InputType2 >::value &&
 				!is_object< OutputType >::value,
-			void >::type * = NULL
+			void > * = NULL
```
Comment (djelovina) [inline include/alp/reference/blas0.hpp:76]: use `nullptr` instead of `NULL` (also in all places below)

**Review by djelovina — APPROVED** [review]

File: include/alp/base/blas0.hpp (hunk 2 above)
Comment (hidanielesgit) [reply include/alp/base/blas0.hpp:432]: (same here)

File: include/alp/amf-based/storagebasedmatrix.hpp (hunk 4 above)
Comment (hidanielesgit) [reply include/alp/amf-based/storagebasedmatrix.hpp]: Cannot use default in template friend declaration

**Pull request merged**